modules/bwa.nf

```diff
@@ -0,0 +1,56 @@
+process bwa_index {
+  label 'bwa'
+  
+  input:
+    path(fasta)
+  
+  output:
+    path(bwa) , emit: index
+  
+  script:
+  """
+  mkdir bwa
+  bwa \\
+    index \\
+    -p bwa/db \\
+    $fasta
+  """
+  
+  stub:
+  """
+  mkdir bwa
+  
+  touch bwa/db.amb
+  touch bwa/db.ann
+  touch bwa/db.bwt
+  touch bwa/db.pac
+  touch bwa/db.sa
+  """
+}
+
+process bwa {
+  label 'bwa'
+
+  input: 
+  tuple val(name), path(input)
+  path(db_index)
+  path(db)
+
+
+  output:
+  tuple val(name), val('raw'), path("${name}.bam"), emit: bam // input just for naming
+
+  script:
+  """
+  INDEX=`find -L ./ -name "*.amb" | sed 's/\\.amb\$//'`
```

Comment:
Isn't the index passed in as a parameter (db_index)? If you only want to strip off the `.amb` you can use something like:
```
index_root=${db_index%.amb}
```
(not tested)
see https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

Reply:
The BWA processes are copied from nf-core 😇 

The index process outputs a folder called `bwa`, and with this line, you find the folder. Maybe `amb` is BWA specific? 🤔